x-pack/plugins/cases/public/components/cases_context/index.tsx

```diff
@@ -69,7 +73,14 @@ export const CasesProvider: React.FC<{ value: CasesContextProps }> = ({
   }));
 
   /**
-   * `userCanCrud` prop may change by the parent plugin.
+   * Only update the context if the nested permissions fields changed, this avoids a rerender when the object's reference
+   * changes.
+   */
+  useDeepCompareEffect(() => {
```

Comment:
Hey, yes this will prevent the re-render when the `permisions` object reference changes but the `all` and `read` values stay the same. 
Another option is to keep using `useEffect` and set `[permissions.all, permissions.read]` as dependencies. Both ways should work 👍

Reply:
@semd Do you think we should combine the two useEffects two avoid a second rerender if `appTitle`, `appId`, and `permission` change at the same time?

Reply:
Thanks, @semd! @jonathan-buttner, could you please merge them into one? You can use the old `useEffect` and add to the permission array `permissions.all` and `permissions.read`.